src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -189,11 +189,32 @@ private function flatten(array $array, array &$result, $keySeparator, $parentKey
             if (\is_array($value)) {
                 $this->flatten($value, $result, $keySeparator, $parentKey.$key.$keySeparator);
             } else {
-                $result[$parentKey.$key] = $value;
+                $result[$parentKey.$key] = $this->cleanValueByType($value);
```

Comment:
I would remove the private method and use a ternary operator here directly.

Reply:
We could do it like this:

`$result[$parentKey.$key] = ($value === false) ? 0 : $value;`

Because currently fputcsv will handle the true/1 conversion. Or should we explicitly handle that too in order to be consistent? 

Something like:

`$result[$parentKey.$key] = ($value === false) ? 0 : (($value === true) ? 1 : $value);`

Which do you prefer?

Reply:
I would prefer the explicitness of the second one (where you can remove extra parentheses and put the value first: `$result[$parentKey.$key] = false === $value ? 0 : (true === $value ? 1 : $value);`).